src/python/WMCore/Database/CMSCouch.py

```diff
@@ -1055,7 +1072,12 @@ def replicate(self, source, destination, continuous=False,
             data["filter"] = filter
             if query_params:
                 data["query_params"] = query_params
-        return self.post('/_replicator', data)
+        resp = self.post('/_replicator', data)
+        # If replication is triggered in unit tests, make sure to give it a few
+        # seconds before proceeding with tests (new functionality in CouchDB 3.x)
+        if "unittest" in source or "unittest" in destination:
```

Comment:
Actually not. Between the creation of the document in CouchDB and its actual scheduling as an active task, there is a gap of a few seconds. That's why we have to sleep for a while when creating replications in our unit tests.

For the record, I tested this with different number of seconds and 5secs is the bare minimum to be >=99% sure that the replication task will exist.

Reply:
@amaltaro , if this is the case the code still requires changes. What if it would be delay for 6 seconds. Instead, please make for loop with configurable parameter to retries number of times, e.g. 3 times, which will query back the CouchDB to see if the process (replication) is finished. Then, record into the log each attempt that will give more information while debugging the issue. For instance, it may be network hick-up, etc.

Reply:
I'm not adding more complication to this PR by probing CouchDB and checking whether the replication kicked in. I want to have the BARE minimum amount of changes to move to CouchDB 3.x.

Having a parameter for the sleep time is not too bad though. If it requires only a few lines to be changed, I might add it to this PR soon. Thanks for this suggestion.